tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -1210,15 +1135,15 @@ class SwiftDocumentStructureWalker: public ide::SyntaxModelWalker {
         Node.Kind != SyntaxStructureKind::LocalVariable &&
         Node.Kind != SyntaxStructureKind::GenericTypeParam) {
       if (auto *VD = dyn_cast_or_null<ValueDecl>(Node.Dcl)) {
-        AccessLevel = getAccessLevelUID(inferAccessLevel(VD));
+        AccessLevel = getAccessLevelUID(VD->getFormalAccess());
       } else if (auto *ED = dyn_cast_or_null<ExtensionDecl>(Node.Dcl)) {
-        if (auto StrictAccess = getAccessLevelStrictly(ED))
-          AccessLevel = getAccessLevelUID(StrictAccess.getValue());
+        auto StrictAccess = ED->getDefaultAccessLevel();
+        AccessLevel = getAccessLevelUID(StrictAccess);
       }
       if (auto *ASD = dyn_cast_or_null<AbstractStorageDecl>(Node.Dcl)) {
-        Optional<swift::AccessLevel> SetAccess = inferSetterAccessLevel(ASD);
-        if (SetAccess.hasValue()) {
-          SetterAccessLevel = getAccessLevelUID(SetAccess.getValue());
+        if (ASD->isSettable(/*UseDC=*/nullptr)) {
+          swift::AccessLevel SetAccess = ASD->getSetterFormalAccess();
+          SetterAccessLevel = getAccessLevelUID(SetAccess);
```

Comment:
This change seems like a bad idea. I like being more in sync with the real behavior, but it's also more brittle in the face of invalid code. This code really should be using `hasAccess` because it must not assert.